GitUI/Theming/SystemBrushOverrides.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace GitUI.Theming
+{
+    internal class SystemBrushOverrides
+    {
+        private readonly Dictionary<int, HandleRef> _cache = new();
+
+        public IntPtr GetBrush(int colorref)
+        {
+            if (!_cache.TryGetValue(colorref, out var handle))
+            {
+                var hbrush = NativeMethods.CreateSolidBrush(colorref);
```

Comment:
❗ `HBRUSH` must be disposed of when we no longer need it (e.g. when we uninstall the hook), otherwise we'll leak native handles.
https://docs.microsoft.com/windows/win32/api/wingdi/nf-wingdi-createsolidbrush#remarks

This is how we've wrapped (some) native handled in dotnet/winforms, may be you find it useful:
* https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms.Primitives/src/Interop/Gdi32/Interop.HBRUSH.cs
* https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms.Primitives/src/Interop/Gdi32/Interop.CreateBrushScope.cs